File: benchmark/constant_numerator_division/constant_numerator_division.go
package main

import "fmt"

func main() {
	var iterations int64 = 8000000
	var numerator int64 = 123456789
	var sum int64 = 0

	for i := int64(0); i < iterations; i++ {
		denom := (i % 997) + 1
		sum += (numerator / denom) + (numerator % denom)
	}

	fmt.Printf("Constant numerator: %d\n", sum)
}

File: benchmark/constant_numerator_division/constant_numerator_division.js
const iterations = 8000000;
const numerator = 123456789;
let sum = 0;

for (let i = 0; i < iterations; i++) {
  const denom = (i % 997) + 1;
  sum += Math.trunc(numerator / denom) + (numerator % denom);
}

console.log(`Constant numerator: ${sum}`);
